mmv1/third_party/validator/service_account.go

```diff
@@ -0,0 +1,146 @@
+// ----------------------------------------------------------------------------
+//
+//	This file is copied here by Magic Modules. The code for building up a
+//	storage bucket object is copied from the manually implemented
+//	provider file:
+//	third_party/terraform/resources/resource_storage_bucket.go
+//
+// ----------------------------------------------------------------------------
+package google
+
+import (
+	"fmt"
+	"reflect"
+)
+
+const ServiceAccountAssetType string = "iam.googleapis.com/ServiceAccount"
+
+func resourceConverterServiceAccount() ResourceConverter {
+	return ResourceConverter{
+		AssetType: ServiceAccountAssetType,
+		Convert:   GetServiceAccountCaiObject,
+	}
+}
+
+func GetServiceAccountCaiObject(d TerraformResourceData, config *Config) ([]Asset, error) {
+	name, err := assetName(d, config, "//iam.googleapis.com/projects/{{project}}/serviceAccounts/{{unique_id}}")
+	if err != nil {
+		return []Asset{}, err
+	}
+	if obj, err := GetServiceAccountApiObject(d, config); err == nil {
+		return []Asset{{
+			Name: name,
+			Type: ServiceAccountAssetType,
+			Resource: &AssetResource{
+				Version:              "v1",
+				DiscoveryDocumentURI: "https://iam.googleapis.com/$discovery/rest",
+				DiscoveryName:        "ServiceAccount",
+				Data:                 obj,
+			},
+		}}, nil
+	} else {
+		return []Asset{}, err
+	}
+}
+
+func GetServiceAccountApiObject(d TerraformResourceData, config *Config) (map[string]interface{}, error) {
+	obj := make(map[string]interface{})
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return nil, err
+	}
+
+	descriptionProp, err := expandServiceAccountDescription(d.Get("description"), d, config)
+	if err != nil {
+		return nil, err
+	} else if v, ok := d.GetOkExists("description"); !isEmptyValue(reflect.ValueOf(descriptionProp)) && (ok || !reflect.DeepEqual(v, descriptionProp)) {
+		obj["description"] = descriptionProp
+	}
+
+	emailProp, err := expandServiceAccountDescription(d.Get("email"), d, config)
+	if err != nil {
+		return nil, err
+	} else if v, ok := d.GetOkExists("email"); !isEmptyValue(reflect.ValueOf(emailProp)) && (ok || !reflect.DeepEqual(v, emailProp)) {
+		obj["email"] = emailProp
+	}
+
+	displayNameProp, err := expandServiceAccountDisplayName(d.Get("display_name"), d, config)
+	if err != nil {
+		return nil, err
+	} else if v, ok := d.GetOkExists("display_name"); !isEmptyValue(reflect.ValueOf(displayNameProp)) && (ok || !reflect.DeepEqual(v, displayNameProp)) {
+		obj["displayName"] = displayNameProp
+	}
+
+	nameProp, err := expandServiceAccountName(d.Get("name"), d, config)
```

Comment:
similar to the email above, since name is an exported field, not a supported argument.

Reply:
unique_id as well